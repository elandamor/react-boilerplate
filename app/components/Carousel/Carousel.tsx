import React from 'react';
import classNames from 'classnames';
// Components
import Icon from '../Icon';
// Styles
import Wrapper from './styles';

function Arrow(props) {
  const { direction, onClick } = props;

  return (
    <div
      className={classNames(
        'c-arrow',
        direction === 'next' ? 'c-arrow--next' : 'c-arrow--prev',
      )}
      onClick={onClick}
      role="button"
      tabIndex={0}
    >
      <Icon icon="add" />
    </div>
  );
}

const defaultSettings = {
  nextArrow: <Arrow direction="next" />,
  prevArrow: <Arrow direction="prev" />,
  slidesToScroll: 1,
  slidesToShow: 1,
  speed: 300,
};

/**
 * @render react
 * @name Carousel component
 * @description Carousel component.
 * @example
 * <Carousel />
 */

interface IProps {
  className?: string;
  settings?: object;
}

class Carousel extends React.Component<IProps, {}> {
  public render() {
    const { className, settings } = this.props;

    return (
      <Wrapper
        className={classNames('c-carousel', className)}
        {...defaultSettings}
        {...settings}
      >
        {this.props.children}
      </Wrapper>
    );
  }
}

export default Carousel;
