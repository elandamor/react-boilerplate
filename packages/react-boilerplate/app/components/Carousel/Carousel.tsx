import React from 'react';
import classNames from 'classnames';
// Components
import Icon from '../Icon';
// Styles
import Wrapper from './styles';

interface IArrowProps {
  direction?: string;
  icon: string;
  onClick?: () => void;
}

const Arrow = (props: IArrowProps) => {
  const { direction, icon, onClick } = props;

  return (
    <div
      className={classNames(
        'c-arrow',
        direction === 'next' ? 'c-arrow--next' : 'c-arrow--prev',
      )}
      onClick={onClick}
      role="button"
      tabIndex={0}
    >
      <Icon icon={icon} />
    </div>
  );
};

const defaultSettings = {
  nextArrow: <Arrow direction="next" icon="next" />,
  prevArrow: <Arrow direction="prev" icon="previous" />,
  slidesToScroll: 1,
  slidesToShow: 1,
  speed: 300,
};

/**
 * @render react
 * @name Carousel component
 * @description Carousel component.
 * @example
 * <Carousel />
 */

interface IProps {
  className?: string;
  settings?: object;
}

class Carousel extends React.Component<IProps, {}> {
  public render() {
    const { className, settings } = this.props;

    return (
      <Wrapper
        className={classNames('c-carousel', className)}
        {...defaultSettings}
        {...settings}
      >
        {this.props.children}
      </Wrapper>
    );
  }
}

export default Carousel;
