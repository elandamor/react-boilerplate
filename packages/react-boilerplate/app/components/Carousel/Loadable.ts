/**
 *
 * Asynchronously loads the component for Carousel
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
