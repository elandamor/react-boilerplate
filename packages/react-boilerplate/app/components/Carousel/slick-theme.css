/* Arrows */
.slick-prev,
.slick-next {
  background: transparent;
  border: none;
  color: transparent;
  cursor: pointer;
  display: block;
  height: 20px;
  line-height: 0;
  outline: none;
  padding: 0;
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  width: 20px;
}

.slick-prev:hover,
.slick-prev:focus,
.slick-next:hover,
.slick-next:focus {
  background: transparent;
  color: transparent;
  outline: none;
}

.slick-prev:hover:before,
.slick-prev:focus:before,
.slick-next:hover:before,
.slick-next:focus:before {
  opacity: 1;
}

.slick-prev.slick-disabled:before,
.slick-next.slick-disabled:before {
  opacity: 0.25;
}

.slick-prev:before,
.slick-next:before {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  color: #ffffff;
  font-size: 20px;
  line-height: 1;
  opacity: 0.75;
}

.slick-prev {
  left: -25px;
}

[dir='rtl'] .slick-prev {
  left: auto;
  right: -25px;
}

.slick-prev:before {
  content: '←';
}

[dir='rtl'] .slick-prev:before {
  content: '→';
}

.slick-next {
  right: -25px;
}

[dir='rtl'] .slick-next {
  right: auto;
  left: -25px;
}

.slick-next:before {
  content: '→';
}

[dir='rtl'] .slick-next:before {
  content: '←';
}

/* Dots */
.slick-dotted.slick-slider {
  margin-bottom: 30px;
}

.slick-dots {
  bottom: -25px;
  display: block;
  list-style: none;
  margin: 0;
  padding: 0;
  position: absolute;
  text-align: center;
  width: 100%;
}

.slick-dots li {
  cursor: pointer;
  display: inline-block;
  height: 20px;
  margin: 0 5px;
  padding: 0;
  position: relative;
  width: 20px;
}

.slick-dots li button {
  background: transparent;
  border: 0;
  color: transparent;
  cursor: pointer;
  display: block;
  font-size: 0;
  height: 20px;
  line-height: 0;
  outline: none;
  padding: 5px;
  width: 20px;
}

.slick-dots li button:hover,
.slick-dots li button:focus {
  outline: none;
}

.slick-dots li button:hover:before,
.slick-dots li button:focus:before {
  opacity: 1;
}

.slick-dots li button:before {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  color: #000000;
  content: '•';
  height: 20px;
  left: 0;
  line-height: 1;
  opacity: 0.25;
  position: absolute;
  text-align: center;
  top: 0;
  width: 20px;
}

.slick-dots li.slick-active button:before {
  color: #000000;
  opacity: 0.75;
}
