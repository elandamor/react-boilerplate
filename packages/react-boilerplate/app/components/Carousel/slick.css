.slick-slider {
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  display: block;
  position: relative;
  touch-action: pan-y;
  user-select: none;
}

.slick-list {
  display: block;
  margin: 0;
  overflow: hidden;
  padding: 0;
  position: relative;
}

.slick-list:focus {
  outline: none;
}

.slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.slick-slider .slick-track,
.slick-slider .slick-list {
  transform: translate3d(0, 0, 0);
}

.slick-track {
  display: block;
  position: relative;
  left: 0;
  margin-left: auto;
  margin-right: auto;
  top: 0;
}

.slick-track:before,
.slick-track:after {
  content: "";
  display: table;

}

.slick-track:after {
  clear: both;
}
.slick-loading .slick-track {
  visibility: hidden;
}

.slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

[dir="rtl"] .slick-slide {
  float: right;
}

.slick-slide img {
  display: block;
}

.slick-slide.slick-loading img {
  display: none;
}

.slick-slide.dragging img {
  pointer-events: none;
}

.slick-initialized .slick-slide {
  display: block;
}

.slick-loading .slick-slide {
  visibility: hidden;
}

.slick-vertical .slick-slide {
  border: 1px solid transparent;
  display: block;
  height: auto;

}

.slick-arrow.slick-hidden {
  display: none;
}
