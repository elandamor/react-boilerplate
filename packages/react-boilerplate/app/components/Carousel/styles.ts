import styled from 'styled-components';
import Carousel from 'react-slick';

import './slick.css';
import './slick-theme.css';

const Wrapper = styled(Carousel)`
  opacity: 0;
  outline: none;
  position: relative;

  &.slick-initialized {
    opacity: 1;
  }

  .c-arrow {
    align-items: center;
    bottom: -48px;
    cursor: pointer;
    display: flex;
    height: 64px;
    outline: none;
    position: absolute;
    width: 48px;
    z-index: 0;

    &:before {
      display: none;
    }

    .icon,
    .icon > svg {
      height: 20px;
      width: 20px;
    }

    &--prev {
      left: 12px;
    }

    &--next {
      right: 12px;
      transform: rotate(180deg);
    }
  }

  .slick-dots {
    bottom: -48px;

    li button::before {
      font-size: 12px;
    }
  }
`;

export default Wrapper;
