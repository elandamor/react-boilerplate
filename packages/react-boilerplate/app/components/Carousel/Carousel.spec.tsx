// Carousel.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Carousel from './index';

describe('<Carousel />', () => {
  it('should display a Carousel', () => {
    const component = renderer
      .create(
        <Carousel>
          <div>Slide 1</div>
          <div>Slide 2</div>
          <div>Slide 3</div>
        </Carousel>,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
