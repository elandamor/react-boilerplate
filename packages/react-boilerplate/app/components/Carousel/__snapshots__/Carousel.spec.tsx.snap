// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> should display a Carousel 1`] = `
.c1 {
  margin: 0 auto;
  position: relative;
}

.c1 svg {
  vertical-align: inherit;
}

.c0 {
  opacity: 0;
  outline: none;
  position: relative;
}

.c0.slick-initialized {
  opacity: 1;
}

.c0 .c-arrow {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  bottom: -48px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  outline: none;
  position: absolute;
  width: 48px;
  z-index: 0;
}

.c0 .c-arrow:before {
  display: none;
}

.c0 .c-arrow .icon,
.c0 .c-arrow .icon > svg {
  height: 20px;
  width: 20px;
}

.c0 .c-arrow--prev {
  left: 12px;
}

.c0 .c-arrow--next {
  right: 12px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c0 .slick-dots {
  bottom: -48px;
}

.c0 .slick-dots li button::before {
  font-size: 12px;
}

<div
  className="slick-slider c-carousel c0 slick-initialized"
  dir="ltr"
>
  <div
    className="c-arrow c-arrow--prev"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <i
      aria-hidden="true"
      className="c-icon c1"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      />
    </i>
  </div>
  <div
    className="slick-list"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={null}
    onMouseMove={null}
    onMouseUp={[Function]}
    onTouchCancel={null}
    onTouchEnd={[Function]}
    onTouchMove={null}
    onTouchStart={[Function]}
    style={Object {}}
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    >
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={-1}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 3
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 1
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide"
        data-index={1}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 2
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide"
        data-index={2}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 3
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={3}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 1
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={4}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 2
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={5}
        onClick={[Function]}
        style={
          Object {
            "width": 0,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Slide 3
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c-arrow c-arrow--next"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <i
      aria-hidden="true"
      className="c-icon c1"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      />
    </i>
  </div>
</div>
`;
